repose-aggregator/components/filters/tenant-culling-filter/src/test/scala/org/openrepose/filters/tenantculling/TenantCullingFilterTest.scala

```diff
@@ -0,0 +1,175 @@
+      forwardedRequest.getSplittableHeaderScala(TENANT_ID) should contain only "123456"
+    }
+
+    it("should cull correctly when there are multiple tenants and one role that match") {
+      val request = new MockHttpServletRequest()
+      addheaders(request)
+      request.addHeader(RELEVANT_ROLES, "banana,baz")
+      val filterChain = new MockFilterChain()
+
+      filter.doFilter(request, mock[HttpServletResponse], filterChain)
+      val forwardedRequest = filterChain.getRequest.asInstanceOf[HttpServletRequestWrapper]
+      val resultMap = unpackMap(forwardedRequest.getHeader(TENANT_ROLES_MAP))
+
+      resultMap should contain ("123456" -> Set("foo", "bar", "baz"))
+      resultMap should contain ("456789" -> Set("baz"))
+      resultMap should not contain ("789012" -> Set("wizard"))
+      forwardedRequest.getSplittableHeaderScala(TENANT_ID) should contain ("123456")
+      forwardedRequest.getSplittableHeaderScala(TENANT_ID) should contain ("456789")
+      forwardedRequest.getSplittableHeaderScala(TENANT_ID) should not contain "789012"
+    }
+
+    it("should cull correctly when there are multiple tenants and multiple roles that match") {
+      val request = new MockHttpServletRequest()
+      addheaders(request)
+      request.addHeader(RELEVANT_ROLES, "foo,wizard")
+      val filterChain = new MockFilterChain()
+
+      filter.doFilter(request, mock[HttpServletResponse], filterChain)
+      val forwardedRequest = filterChain.getRequest.asInstanceOf[HttpServletRequestWrapper]
+      val resultMap = unpackMap(forwardedRequest.getHeader(TENANT_ROLES_MAP))
+
+      resultMap should contain ("123456" -> Set("foo", "bar", "baz"))
+      resultMap should not contain ("456789" -> Set("baz"))
+      resultMap should contain ("789012" -> Set("wizard"))
+      forwardedRequest.getSplittableHeaderScala(TENANT_ID) should contain ("123456")
+      forwardedRequest.getSplittableHeaderScala(TENANT_ID) should not contain "456789"
+      forwardedRequest.getSplittableHeaderScala(TENANT_ID) should contain ("789012")
+    }
+
+    it("should cull correctly when there are multiple tenants and no roles that match") {
+      val request = new MockHttpServletRequest()
+      addheaders(request)
+      request.addHeader(RELEVANT_ROLES, "banana,shazbot")
+      val filterChain = new MockFilterChain()
+
+      filter.doFilter(request, mock[HttpServletResponse], filterChain)
+      val forwardedRequest = filterChain.getRequest.asInstanceOf[HttpServletRequestWrapper]
+      val resultMap = unpackMap(forwardedRequest.getHeader(TENANT_ROLES_MAP))
+
+      resultMap shouldBe empty
+      forwardedRequest.getSplittableHeaderScala(TENANT_ID) shouldBe empty
+    }
+
+    it("should cull correctly when the relevant roles are qualified") {
+      val request = new MockHttpServletRequest()
+      addheaders(request)
+      request.addHeader(RELEVANT_ROLES, "bar/{tenant1}, banana/{tenant2}")
```

Comment:
These will always just be `{X-TENANT}` coming from API Checker/Validator.
However, it doesn't effect the test because anything after the `/` is ignored anyway.

Reply:
I guess it's not much of a coincidence, but I just asked about this earlier today!

To be clear, though, it's not _always_ `{X-TENANT}`, is it? It could be `{X-Tenant-Id}` or whatever else you might set the WADL param name to, right?

Reply:
This is not based on the header's name, it is the API-Checker feature which sets the value. So, yes, it will ___always___ be `{X-TENANT}` for this feature.

Reply:
No, I don't think that is correct. I mean, it will _*always*_ be `{X-TENANT}` if the WADL param is set to `X-TENANT`, but generally, it will just match the WADL param name. And I'm pretty sure the WADL param name must match the header name.

Reply:
Yep. I stand corrected on the ___always___ claim, but for some reason the param name must be in all caps.